packages/next/export/worker.js

```diff
@@ -28,7 +28,8 @@ process.on(
         await sema.acquire()
         const { page, query = {} } = exportPathMap[path]
         const req = { url: path }
-        const res = {}
+        const res = path === '/404.html' ? { statusCode: 404 } : {}
```

Comment:
The proposed solution in #6243 didn't work since `err` wasn't actually null when exporting. I was a bit worried defaulting to 404 for _any_ scenario directly in `_error.js` would have the potential to bite us further down the line, so maybe this might work as an alternative?